Footer

Deafveloper opened the pull request:
-Created footer and add style
-Created icon folder and added icon files
-Created Social links file

linked issue[https://github.com/signlanguagetech/sign-tech-interview/issues/33](url)

## Summary by Sourcery

Implement a comprehensive footer with social links and enhanced styling for the website

New Features:
- Added social media links with icons for YouTube, Twitter, GitHub, and LinkedIn
- Included a contact email link in the footer

Enhancements:
- Redesigned footer with responsive layout
- Added copyright notice with dynamic year
- Improved footer styling with gradient background and hover effects

Documentation:
- Added screen reader text for social media links

Deafveloper committed:
add footer layout and styling

created footer and add styling

Deafveloper committed:
Updated Footer.astro

added link to privacy policy

miladvafaeifard commented on src/components/SocialLinks.astro:
```diff
@@ -0,0 +1,24 @@
+---
+import YoututbeIcon from './icons/YoutubeIcon.astro';
```
i would follow the naming convention for file names: Youtubeicon.astro -> YouTubeIcon.astro

miladvafaeifard commented on src/components/SocialLinks.astro:
```diff
@@ -0,0 +1,24 @@
+---
+
+<a href='https://www.youtube.com/@SignLanguageTech' target='_blank'>
+  <span class='sr-only'>Sign Language Tech YouTube</span>
```
is 'sr-only' needed? i can't see this class css from the styles

Deafveloper replied:
yes, for screen readers. to help people recongnize the icons in footer. I forgot to add the css style to file

tech-andgar commented on src/components/icons/Linkedinicon.astro:
'Linkedinicon.astro' to 'LinkedinIcon.astro'

tech-andgar commented on src/components/icons/Youtubeicon.astro:
https://github.com/signlanguagetech/sign-tech-interview/pull/94/files#r2082560602

tech-andgar replied:
https://github.com/signlanguagetech/sign-tech-interview/pull/94/files#r2082433781

tech-andgar commented:
You need to review your local build. 
![image](https://github.com/user-attachments/assets/276d06c4-1974-4f61-b81e-23c3aa382dbd)

Deafveloper commented on src/components/overrides/Footer.astro:
```diff
@@ -1,22 +1,78 @@
+    <div>
+      <p>© {today.getFullYear()} Sign Language Tech. Interview All rights reserved.</p>
+      <div class="social-icons">
+        <SocialLinks />
```
done

Deafveloper committed:
Summary updated footer and social links based on review feedback

- Added rel="noopener noreferrer" to all target="_blank" links for security

- Standardized import and filename casing for YouTube and LinkedIn icons

- Added .sr-only CSS Style for screen reader accessibility

- Fixed typo in LinkedIn screen reader text

Deafveloper committed:
Updated SocialLinks

revert the capital letter

Deafveloper commented:
> You need to review your local build because the build of project failed. ![image](https://private-user-images.githubusercontent.com/19521054/442336363-276d06c4-1974-4f61-b81e-23c3aa382dbd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcxMDAwMzQsIm5iZiI6MTc0NzA5OTczNCwicGF0aCI6Ii8xOTUyMTA1NC80NDIzMzYzNjMtMjc2ZDA2YzQtMTk3NC00ZjYxLWI4MWUtMjNjM2FhMzgyZGJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTEzVDAxMjg1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiNGNiZDVhYzQyZGUyOTBkODFmZWU0ZDZmNGZlNzdhOWI5ZDA0ZjBhODY5MmNlOWUzOWQyZDM4ZDUyNmM1MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7r7IR_ezAxLPPqbLnfdw91lAghDzfpTxl99A705SiaE)

checking this now

tech-andgar commented:
> > You need to review your local build because the build of project failed. ![image](https://private-user-images.githubusercontent.com/19521054/442336363-276d06c4-1974-4f61-b81e-23c3aa382dbd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcxMDAwMzQsIm5iZiI6MTc0NzA5OTczNCwicGF0aCI6Ii8xOTUyMTA1NC80NDIzMzYzNjMtMjc2ZDA2YzQtMTk3NC00ZjYxLWI4MWUtMjNjM2FhMzgyZGJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTEzVDAxMjg1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiNGNiZDVhYzQyZGUyOTBkODFmZWU0ZDZmNGZlNzdhOWI5ZDA0ZjBhODY5MmNlOWUzOWQyZDM4ZDUyNmM1MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7r7IR_ezAxLPPqbLnfdw91lAghDzfpTxl99A705SiaE)
> 
> checking this now

![image](https://github.com/user-attachments/assets/9701401f-80bb-4b1d-b185-8a50a7b08294)

tech-andgar commented on src/components/SocialLinks.astro:
```diff
@@ -0,0 +1,38 @@
+  <span class='sr-only'>Go to Milad's GitHub</span>
+  <GitHubIcon height='32px' width='32px' />
+</a>
+<a href='https://www.linkedin.com/in/miladvafaeifard' target='_blank' rel='noopener noreferrer'>
```
change to "https://www.linkedin.com/company/sign-language-tech"

tech-andgar commented on src/components/SocialLinks.astro:
```diff
@@ -0,0 +1,38 @@
+  <span class='sr-only'>Follow Me on Twitter</span>
+  <TwitterIcon height='32px' width='32px' />
+</a>
+<a href='https://github.com/miladvafaeifard' target='_blank' rel='noopener noreferrer'>
```
change to "https://github.com/signlanguagetech/sign-tech-interview"

tech-andgar commented on src/components/SocialLinks.astro:
```diff
@@ -0,0 +1,38 @@
+  <span class='sr-only'>Sign Language Tech YouTube</span>
+  <YouTubeIcon height='32px' width='32px' />
+</a>
+<a href='https://twitter.com/MiladVafaeifard' target='_blank' rel='noopener noreferrer'>
```
change to "https://x.com/signlanguagete1"

tech-andgar commented on src/components/SocialLinks.astro:
```diff
@@ -0,0 +1,38 @@
+  <TwitterIcon height='32px' width='32px' />
+</a>
+<a href='https://github.com/miladvafaeifard' target='_blank' rel='noopener noreferrer'>
+  <span class='sr-only'>Go to Milad's GitHub</span>
```
Change to "Sign Language Tech"

tech-andgar commented on src/components/SocialLinks.astro:
```diff
@@ -0,0 +1,38 @@
+  <GitHubIcon height='32px' width='32px' />
+</a>
+<a href='https://www.linkedin.com/in/miladvafaeifard' target='_blank' rel='noopener noreferrer'>
+  <span class='sr-only'>Go to Milad's LinkedIn</span>
```
Change to "Sign Language Tech"

Deafveloper commented:
![Screenshot 2025-05-13 at 11 29 49 AM](https://github.com/user-attachments/assets/2d093fdb-93f9-4f7d-a6c9-9b1bd1c1f738)
I made the edits and pushed it. I am unsure why it didn't update file name changes.

Deafveloper committed:
Summary

Please Ignore. Purposely renaming files wrong

Deafveloper committed:
Merge branch 'signlanguagetech:main' into main

Deafveloper committed:
Summery

Updated correct file names under icon folder

miladvafaeifard commented:
Hi @Deafveloper !

my apologises that i didn't mention them before, actually, it is much better to use the built-in component in the starlight framework following:
```ts
import { Icon } from "@astrojs/starlight/components";

<Icon name="twitter" />
<Icon name="linkedin" />
<Icon name="youtube" />
<Icon name="github" />

```
